### ISSUE TITLE
Some system shortcuts don't work on macos (Cmd+Q, Cmd+,)

### ISSUE THREAD

**Issue by Coops0** [zen-browser/desktop#1629]
### Captchas

- [x] I have read the instructions.
- [X] I have searched existing issues and avoided creating duplicates.
- [X] I am not filing an enhancement request.

### What happened?

OLD:
*On Mac (Sequoia), holding the Cmd key and pressing a number 1-9 usually takes you to that tab index. Now, it just does nothing. I am still able to `ctrl+tab` through them, and other cmd shortcuts like `Cmd+T` and `Cmd+C `still work. *

CURRENTLY:
Some keybinds do not work, such as `Cmd+Q`, `Cmd+,` (settings), and `Opt + Cmd + U` (split tab).

### Reproducible?

- [X] I have checked that this issue cannot be reproduced on Mozilla Firefox.

### Version

1.0.1-a.9

### What platform are you seeing the problem on?

macOS - aarch64

**Comment by stavros-k** [comment]
Other broken shortcuts:
- Ctrl+Shift+T (Restore tab)
- Ctrl+R (Reload)

**Comment by stavros-k** [comment]
> Can confirm all of this - it is very strange ctrl works for these...

It's just the windows/linux shortcuts, which I assume are the defaults until a mac is identified or something.

**Comment by netzzwerk** [comment]
Also happening on macOS Sonoma 14.6.1

**Comment by say4n** [comment]
Possibly caused by: https://github.com/zen-browser/desktop/commit/232fa448ca4177af0cf5d961e385ab2fd40785d1? Seems to pertain to keyboard shortcuts and is tagged as the latest release.

**Comment by AleksZakirov** [comment]
> Other broken shortcuts:
> 
>     * `Cmd`+`Shift`+`T` (Restore tab)
> 
>     * `Cmd`+`R` (Reload)
> 
>     * `Cmd`+`Shift`+`L` (Bitwarden autofil trigger)
> 
>     * `Cmd` + `A` (Select All)
> 
> 
> Note that the above works if I use the `Ctrl` instead of `Cmd` So something is funky there

for me Cmd+A, Cmd+T work, Cmd+R and Cmd+Shift+T don't.

**Comment by szelonka** [comment]
Same issue here, I'm on macOS Sequoia 15.0 (24A335) 

I started to use Zen today, so IDK if I'm missing some other shortcuts, but at least these ones failed:
- `cmd+shift+t` to restore last tab closed (is working with `ctrl` instead)
- `cmd+r`
- ``cmd+` `` (this is to use 1password autofill)

**Comment by sebazelonka** [comment]
Same issue here, I'm on macOS Sequoia 15.0 (24A335) 

I started to use Zen today, so IDK if I'm missing some other shortcuts, but at least these ones failed:
- `cmd+shift+t` to restore last tab closed (is working with `ctrl` instead)
- `cmd+r`
- ``cmd+` `` (this is to use 1password autofill)

**Comment by mauro-balades** [comment]
Will fix this ASAp, releasing today.

**Comment by HarryHeres** [comment]
Apologies to everyone. We included a patch for macOS specifically adjusting functionality of the modifiers and it seems like it is interfering with the previous functionality. As @mauro-balades noted, we'll fix this with the new release.

**Comment by bangoim** [comment]
> Other broken shortcuts:
> 
>     * `Cmd`+`Shift`+`T` (Restore tab)
> 
>     * `Cmd`+`R` (Reload)
> 
>     * `Cmd`+`Shift`+`L` (Bitwarden autofil trigger)
> 
>     * `Cmd` + `A` (Select All)
> 
> 
> Note that the above works if I use the `Ctrl` instead of `Cmd` So something is funky there

For me, `Cmd`+`R` and `Cmd`+`A` are working. But would like to add `Cmd`+`L` to the list of non-working shortcuts.

**Comment by Coops0** [comment]
Unfortunately, after updating, the Cmd+<Number> keybind still doesn't work. Some others do seem to work though.
Cmd+, (settings) doesn't seem to work either? 

When on the settings tab, other shortcuts do not work now either?? e.x. Cmd+W
Ctrl+<number> doesn't work either

**Comment by Valadramuz** [comment]
Hi!
I don't know if I need to open a new issue but on the latest version _1.0.1-a.4_ only the following commands work:

- CMD + L
- CMD + R
- CMD + T
- CMD + W
- CMD + P
- CMD + +
- CMD + -
- CMD + 0

The commands that DO NOT work are:

- CMD + Number
- CTRL + Number (worked on _1.0.1-a.3_ when 'CMD + Number' didn't work)
- Any user keyboard shortcut with CMD
- Whenever you try to modify a keyboard shortcut: 'CTRL + OPTION + E' you get 'CTRL + OPTION + €' (this has been an issue since previous versions)

**Comment by y2k04** [comment]
Ctrl+W doesn't work anymore for me. It used to be able to before I restarted the browser.

**Comment by HarryHeres** [comment]
So after the update to 1.0.1-a.4, some of the `CMD` shortcuts work but some of the still don't, is that right?

**Comment by josefssonemil** [comment]
> So after the update to 1.0.1-a.4, some of the `CMD` shortcuts work but some of the still don't, is that right?

Correct, 1.0.1-a4 has the behavior that @Valadramuz listed

**Comment by KarimElsayad247** [comment]
https://github.com/zen-browser/desktop/issues/1686

`CTRL + Shift + B` does (bookmarks bar toggle) not work as well.

**Comment by KarimElsayad247** [comment]
@HarryHeres I'm facing something similar with linux, with `ctrl + shift + b` not working. Could this be related to this max issue?

**Comment by bangoim** [comment]
Would like to add that in the in the release notes is said that it was added a new system for handling keyboard shortcuts. The interface is the same, didn't see no difference and it is loaded with duplicates and swapped commands. Also, in MacOS the shortcuts that should be showing `Cmd` + something are showing as `Ctrl` + something. 

`Cmd` + `F`: makes browser go full screen instead of searching inside the tab.
`Cmd` + `,`: does not go to settings.
`Cmd` + `{number}`: does not switch tabs.
`Cmd` + `Opt` + `U`: sends to view-source:about:preferences#zenCKS instead of closing split view.
 
 Those are the ones that most impacted me, there is possibly more. 
 
 Another thing is the missing labels on some shortcuts and duplicated ones.
 
<img width="707" alt="image" src="https://github.com/user-attachments/assets/bd95218e-f461-4a16-9479-4114af91bf29">

**Comment by KarimElsayad247** [comment]
I can confirm that `Alt + 1` through `Alt + 9` don't work.

**Comment by kgarrett67** [comment]
Not sure if this should go here, but with version 1.0.1-a.4, I cannot open bookmarks or history using Ctrl + B or Ctrl + H.

**Comment by gk98s** [comment]
I'm on windows and my shortcuts are also all messed up, the open private window changed itself to "ctrl shift p" and the new tab one stopped working as of the latest update.

**Comment by fhfuih** [comment]
To add: I have created multiple profiles before recent updates, and only the main (default) profile got affected.

**Comment by KarimElsayad247** [comment]
How do I update to the recent version? About Zen window doesn't have a way to check for a new version.

**Comment by Amplificator** [comment]
The option for setting a shortcut for "Next tab" and "Previous tab" is gone now - can we have these back?

**Comment by yoghurtlee-thu** [comment]
`Cmd + ,` broke again in 1.0.1-a-6 :(

**Comment by Coops0** [comment]
> `Cmd + ,` broke again in 1.0.1-a-6 :(

Unfortunately can confirm, I also notice `Cmd + q` is broken.

**Comment by bangoim** [comment]
> `Cmd + ,` broke again in 1.0.1-a-6 :(

`Cmd + h` also not working.

**Comment by yoghurtlee-thu** [comment]
The problem is still bothering me in 1.0.1-a-7 :(

**Comment by say4n** [comment]
Seems like 1.0.1-a-5 was the last working version.

**Comment by sebazelonka** [comment]
same here, I'm on 1.0.1-a.7 and `cmd+q` `cmd+,` still broken

**Comment by bangoim** [comment]
From what I tested in 1.0.1-a.7, those below are broken:

- `Cmd + Q`
- `Cmd + ,`
- `Cmd + h`

**Comment by mauro-balades** [comment]
This *might* get fixed in the next release, still testing them out. Thanks!

**Comment by vileflume** [comment]
i was having similar issues and oddly, unsetting keyboard shortcuts in about:preferences allowed me to use the macos defaults (ie cmd + q / h / ,)

**Comment by yoghurtlee-thu** [comment]
Sadly in a.8 the issue still exists, on macOS Sequoia :(

**Comment by yoghurtlee-thu** [comment]
> Sadly in a.8 the issue still exists, on macOS Sequoia :(

After I quit Zen and restart it for 3 times, it works. Strange.

**Comment by Coops0** [comment]
After updating, I had to reset my keybinds and it worked for a little bit, and now it is not?

I do notice some oddities with the split keybinds for example.
Like the Cmd+Opt+U didn't work, until I reassigned it to the same thing. It works, but now shows this
<img width="198" alt="image" src="https://github.com/user-attachments/assets/72370a9b-7b5e-4ebb-bcbe-33961d81b02b">

**Comment by Amplificator** [comment]
The option for setting a shortcut for "Next tab" and "Previous tab" is gone since a few releases back - can we have these back?

**Comment by berealisss** [comment]
1.0.1-a.8 on macOS Sonoma 14.2.1. `Cmd+,` still doesn't work

**Comment by chriskirknielsen** [comment]
Confirming `Cmd+,` does not work on 1.0.1-a.8 with Sonoma 14.6, but it is correctly assigned:
<img width="144" alt="Preferences button tooltip showing the Command Comma shortcut" src="https://github.com/user-attachments/assets/0d93b7be-1912-4cd8-bd7f-1d709ac84ee3">
<img width="766" alt="Shortcut page showing that Open Preferences is assigned to Command Comma" src="https://github.com/user-attachments/assets/77b9ad2f-7746-4323-8709-3bb7c3be4419">

**Comment by ghost** [comment]
I unset `Cmd + ,` and now it works. Strange as a $3 bill.

**Comment by ghost** [comment]
> I unset `Cmd + ,` and now it works. Strange as a $3 bill.

Confirm, that is, in fact, quite interesting. And it works.

Did the same for every command under 'Zen Browser' tab in the menu bar, and now they execute as intended.

**Comment by hsvfan-jan** [comment]
I haven't checked all mentioned shortcuts but Cmd+, and Cmd+Q still don't work on macOS 15.0.1 (24A348) and Zen 15.0.1 (24A348)

**Comment by kuspoes** [comment]
reseting the shortcut setting to default value, solved this problem in my mac. 

⌘+ Q,
⌘+ W,
⌘+ ⇧ + T

is working smoothly

**Comment by mr-cheffy** [comment]
Im working on fixing these. @kuspoes on what version are you on? Do they keep working even after restarting the browser?

**Comment by jonaharagon** [comment]
Can also confirm that on a fresh Zen Browser install going to `about:preferences#zenCKS` and hitting Reset to Default fixes those three shortcuts. They do **not** keep working after restarting the browser for me.

**Comment by bangoim** [comment]
@mauro-balades answering for him. If you reset to default, no. But if you unset them then the commands start working again even after restarting the browser. In fact it has been like that since 1.0.1-a.7.

**Comment by kuspoes** [comment]
@mauro-balades :

> Im working on fixing these. @kuspoes on what version are you on? Do they keep working even after restarting the browser?

I'm currently on version 1.0.1-a.8 and no, after restarting the browser ⌘+ Q is not working again but another shortcut are functioning correctly

**Comment by mr-cheffy** [comment]
Heya!

I was wondering if anyone in macos would be kind enough to download twilight and test out the native commands (CMD+Q, CMD+,, etc.). Thanks!

https://zen-browser.app/download?twilight

**Comment by harryheres** [comment]
> Heya!
> 
> I was wondering if anyone in macos would be kind enough to download twilight and test out the native commands (CMD+Q, CMD+,, etc.). Thanks!
> 
> https://zen-browser.app/download?twilight

Just adding to Mauro - we tested them and they work on a clean profile on latest Sonoma. If you have any issues, please make sure you are not running any programs that could be catching the key presses (like Yabai or AeroSpace) and let us know which version of macOS are you running.

Thank you ❤️

**Comment by andreaswilli** [comment]
Just tested twilight 1.0.1-a.8 (aarch64) on macOS 15.0.1 (24A348) and the shortcuts cmd+q and cmd+, seem to work fine without changing any settings (even with Aerospace).

However, I don't see options to customize key bindings in settings, not sure if this is expected.

**Comment by kuspoes** [comment]
twillight is working like a charms 😀

**Comment by mr-cheffy** [comment]
> However, I don't see options to customize key bindings in settings, not sure if this is expected.

Hmm, all shortcuts dont appear or just the native ones (cmd+Q, etc)?

**Comment by mr-cheffy** [comment]
@kuspoes do shortcuts show up in the preferences page?

**Comment by andreaswilli** [comment]
> Hmm, all shortcuts dont appear or just the native ones (cmd+Q, etc)?

All of them, it looks like this:

<img width="1525" alt="image" src="https://github.com/user-attachments/assets/67c17a7f-cee6-4f61-bafb-38069b4a1066">

**Comment by mr-cheffy** [comment]
Could you send the keyboard-shortcuts.json from your profile's directory?

**Comment by mr-cheffy** [comment]
@HarryHeres this is a fucking mess haha. Some report they work some say they dont even appear on the preferences page...

I was wondering if you have some time later this week/weekend to take a look at them. It's ok if you cant, just wondering

**Comment by KorigamiK** [comment]
> Could you send the keyboard-shortcuts.json from your profile's directory?


@mauro-balades I've attatched my keyboard-shortcuts.json on linux, these shortcuts work only till the version `Mozilla Zen 1.0.1-a.3`. Upgrading to any higher version breaks (almost) all the keybinds that I had previously set and all the zen specific ones as well.
 
[zen-keyboard-shortcuts.json](https://github.com/user-attachments/files/17369592/zen-keyboard-shortcuts.json)

**Comment by KorigamiK** [comment]
I'm also linking issues which might be related to this #1828 #1706

**Comment by KorigamiK** [comment]
@mauro-balades I would highly suggest bisecting to find the changes that caused the regression.

**Comment by mr-cheffy** [comment]
> I've attatched my keyboard-shortcuts.json on linux, these shortcuts work only till the version `Mozilla Zen 1.0.1-a.3`. Upgrading to any higher version breaks (almost) all the keybinds that I had previously set and all the zen specific ones as well.

iirc shortcuts are suffering a rework since that version. That's why some might be missing, since shortcuts in firefox are handled by, and quote, actual "magic voodoo". Im trying to get native ones fixed for macos, then we can start expanding on add new shortcuts.

**Comment by KorigamiK** [comment]
> > I've attatched my keyboard-shortcuts.json on linux, these shortcuts work only till the version `Mozilla Zen 1.0.1-a.3`. Upgrading to any higher version breaks (almost) all the keybinds that I had previously set and all the zen specific ones as well.
> 
> 
> 
> iirc shortcuts are suffering a rework since that version. That's why some might be missing, since shortcuts in firefox are handled by, and quote, actual "magic voodoo". Im trying to get native ones fixed for macos, then we can start expanding on add new shortcuts.

Maybe I'm missing something but since shortcuts used to work perfectly fine before, why is the rework required?

**Comment by andreaswilli** [comment]
> Could you send the keyboard-shortcuts.json from your profile's directory?

`zen-keyboard-shortcuts.json` does not exist in the profile crated by twilight. If I copy it over from my other profile, then shortcuts show up. Clicking Reset to Default gets rid of the file and shortcuts disappear again.

**Comment by ghost** [comment]
> > Could you send the keyboard-shortcuts.json from your profile's directory?
> 
> `zen-keyboard-shortcuts.json` does not exist in the profile crated by twilight. If I copy it over from my other profile, then shortcuts show up. Clicking Reset to Default gets rid of the file and shortcuts disappear again.

Yes, the problem persists in 1.0.1-a.9. I've created a new issue #2106 to specifically discuss and resolve this issue.

**Comment by GunGunGun** [comment]
> > > I've attatched my keyboard-shortcuts.json on linux, these shortcuts work only till the version `Mozilla Zen 1.0.1-a.3`. Upgrading to any higher version breaks (almost) all the keybinds that I had previously set and all the zen specific ones as well.
> > 
> > 
> > iirc shortcuts are suffering a rework since that version. That's why some might be missing, since shortcuts in firefox are handled by, and quote, actual "magic voodoo". Im trying to get native ones fixed for macos, then we can start expanding on add new shortcuts.
> 
> Maybe I'm missing something but since shortcuts used to work perfectly fine before, why is the rework required?

We overcooked it, the old shortcut system had only a few specific shortcuts, but new system had all shortcuts  and that's the issue.

It works on some MacOS system right now, but not on some specific Mac users, so we need testers to fix this completely.

**Comment by harryheres** [comment]
@mauro-balades sure, I'll take a look during this week

**Comment by mr-cheffy** [comment]
Just pushed https://github.com/zen-browser/components/commit/612abc81e78ccdfbdf4d3c0bf6b37e94a87ed1c2. Ill release it to twilight and see how that goes. Taking all of these out at once is gonna be insane

![image](https://github.com/user-attachments/assets/b22cf841-8801-47b4-8d23-0d5a2139ebf5)

**Comment by luigigorlero** [comment]
The latest two alphas have solved most of the issues I personally had with keyboard shortcuts (on latest MacOS 15, Mac Mini M2, German ISO keyboard). I am only unable to close split view with the default `Opt + Cmd + U` as that opens Firefox’s `View Source: Cmd + U`

Thank you for your spectacular work!

**Comment by mr-cheffy** [comment]
Could anyone try the latest twilight release now? Shortcuts should be fixed. Please let me know.

https://zen-browser.app/download?twilight

**Comment by londeril** [comment]
zen-twilight works again on my test VM :)

![2024-10-15-131143_hyprshot](https://github.com/user-attachments/assets/9978ebf8-5a13-40c9-818a-8283cbff0b44)

**Comment by mr-cheffy** [comment]
Nice! I still need verification from any macos user that cmd+Q works

**Comment by mr-cheffy** [comment]
![image](https://github.com/user-attachments/assets/52f29968-6771-4128-a950-36df381bcadf)

oh. my... The keyboard shortcut isnt set as internal... It's always the small stuff

**Comment by mr-cheffy** [comment]
Fixed with https://github.com/zen-browser/components/commit/ecdc196752a0f9a758595b435236383ca8ab2ac6, im so embarrassed haha

**Comment by londeril** [comment]
Nice :) Will you push a new package or should I switch to zen-browser-git for the time being?

**Comment by mr-cheffy** [comment]
ill release today. Ill work a bit more on it for until 7PM (it's 3PM right now) and the build takes 4 hours.

**Comment by londeril** [comment]
Perfect! Thanks so much for your work!

**Comment by chriskirknielsen** [comment]
> Nice! I still need verification from any macos user that cmd+Q works

Tested on macOS 14.4 with Zen Twilight 1.0.1-a.9 (aarch64), `Cmd+,` and `Cmd+Q` right work out of the box (I have never installed Zen Twilight before). FYI, not working on regular Zen a.9 but I suspect that's expected right now, I'll be happy to test the next release for this as well.

EDIT: UH oh. I closed Zen Twilight and re-opened it. The shortcuts don't work any longer.

**Comment by londeril** [comment]
> EDIT: UH oh. I closed Zen Twilight and re-opened it. The shortcuts don't work any longer.

Works for me on Linux - after re-opening Zen and after rebooting the system.

**Comment by mr-cheffy** [comment]
> EDIT: UH oh. I closed Zen Twilight and re-opened it. The shortcuts don't work any longer.

Yeah, i've fixed that

**Comment by mr-cheffy** [comment]
God this is going to feel good.

https://zen-browser.app/release-notes/1.0.1-a.10

**Comment by Coops0** [comment]
can confirm this is fixed!
🎉congrats! thanks for all your hard work!

**Comment by KarimElsayad247** [comment]
Huzzah! It has been done!

**Comment by Amplificator** [comment]
I know I've mentioned it before in this thread, but ever since the re-work of shortcuts started the "Previous tab" and "Next tab" disappeared - any chance of getting those added back again?

If not, I'll be quiet and move on :)

**Comment by mr-cheffy** [comment]
> I know I've mentioned it before in this thread, but ever since the re-work of shortcuts started the "Previous tab" and "Next tab" disappeared - any chance of getting those added back again?

As previously said, firefox is a mess. The new implementation takes all the registrered shortcuts and applies them into something zen understands. Firefox handles next/previous tab in a different way, by implementing event handlers.

Why? No one knows

So it might take a bit of time to fully cover all the shortcuts. Im really sorry. I already have this writen in my linear app, so no need to make an issue about it. Thanks!

To be honest, im just happy we've got macos shortcuts fixed. Got a huge load off my back

**Comment by londeril** [comment]
Just chiming in to confirm that all my shortcuts are back and work as expected! Thanks again for your hard work!

EDITED: spelling

**Comment by BuruY** [comment]
I don't believe MacOS shortcuts are fully fixed. It may be because I have still a Firefox instance installed. But here are my findings 

- The `Compact Mode` shortcuts do not work
- Other Default shortcuts like `Option+Cmd+C` or `Option+Cmd+E` open developer tools for me and not the e.g. `Forward Workspace`

Changing these from their default settings work for some but not all

**Comment by nguyenjessev** [comment]
Yeah, I also still have some issues with CMD+Opt+C to toggle Compact Mode. It works most of the time, but sometimes it breaks and reverts to activating the element picker in dev tools. Haven't been able to pinpoint what's causing it to revert to that behavior.

**Comment by k-soomro** [comment]
I am still facing this issue on Mac Sonoma. Are there any fixes?

**Comment by isaksamsten** [comment]
> I am still facing this issue on Mac Sonoma. Are there any fixes?

Me too. I reassigned the keys, and they are shown as:

<img width="630" alt="Image" src="https://github.com/user-attachments/assets/7a19dd18-c53d-4ef1-9a29-9261e85b596f" />

Now they work properly. Perhaps it has to do with the keyboard layout? I'm using EurKey and US Int. and the same keys are shown.

**Comment by k-soomro** [comment]
> > I am still facing this issue on Mac Sonoma. Are there any fixes?
> 
> Me too. I reassigned the keys, and they are shown as:
> <img alt="Image" width="630" src="https://private-user-images.githubusercontent.com/167128/409541755-7a19dd18-c53d-4ef1-9a29-9261e85b596f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mzg2NzEzNTQsIm5iZiI6MTczODY3MTA1NCwicGF0aCI6Ii8xNjcxMjgvNDA5NTQxNzU1LTdhMTlkZDE4LWM1M2QtNGVmMS05YTI5LTkyNjFlODViNTk2Zi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwMjA0JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDIwNFQxMjEwNTRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT04ZDg2N2MxM2JlODZiODU0ZjkwMTc0ODZiZTdmNmExMDY5YmQ5MTY1OTZlYzJhMDJkYmJjM2Q5OWI1OTE4ZjZiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.RoZSxra81rOyyY0RwYO0LfATjEaeSlxqd7umpwhM6RA">
> 
> Now they work properly. Perhaps it has to do with the keyboard layout? I'm using EurKey and US Int. and the same keys are shown.

I've tried these but they still don't work.

**Comment by prasoon2211** [comment]
I'm on macos Sonoma and shortcuts are a complete mess. Most shortcuts that are shown by default don't work. Remapping them doesn't work either - remapping ends up setting some weird characters in the shortcut. Resetting shortcuts back to default doesn't work either - I restart the browser and it comes back with the same old shortcuts that I had set.

I've spent at least 2 hours trying to figure out a way to fix this. It seems people have been reporting problems with shortcuts on mac for close to a year now it's still completely broken. I'm sorry to say this but looks like Mac is an afterthought for the developers right now.

![Image](https://github.com/user-attachments/assets/51c99883-260d-4557-a9f2-1fb199d0f2c7)
